impl/maven-cli/src/main/java/org/apache/maven/cling/invoker/LookupInvoker.java

```diff
@@ -305,16 +307,29 @@ protected BuildEventListener doDetermineBuildEventListener(C context) {
     }
 
     protected void createTerminal(C context) {
+        boolean stdStreamsRedirected = System.console() == null
+                || context.invokerRequest.in().isPresent()
+                || context.invokerRequest.out().isPresent()
+                || context.invokerRequest.err().isPresent();
+        context.invokerRequest.err().ifPresent(err -> {
+            if (err instanceof PrintStream errPs) {
+                System.setErr(errPs);
+            } else {
+                System.setErr(new PrintStream(err, true));
+            }
+        });
         if (context.terminal == null) {
             MessageUtils.systemInstall(
                     builder -> {
-                        builder.streams(
-                                context.invokerRequest.in().orElse(null),
-                                context.invokerRequest.out().orElse(null));
-                        builder.systemOutput(TerminalBuilder.SystemOutput.ForcedSysOut);
-                        // The exec builder suffers from https://github.com/jline/jline3/issues/1098
-                        // We could re-enable it when fixed to provide support for non-standard architectures,
-                        // for which JLine does not provide any native library.
+                        if (stdStreamsRedirected) {
+                            builder.system(false);
```

Comment:
`system(false)` and `system(true)` are not really needed, this is implicit with the fact that there are streams set.